src/sql/workbench/services/notebook/browser/interface.ts

```diff
@@ -22,12 +22,13 @@ export interface INotebookInput {
 }
 
 export function isINotebookInput(value: any): value is INotebookInput {
-	if (typeof value.defaultKernel === 'object' &&
-		typeof value.connectionProfile === 'object' &&
-		typeof value.isDirty === 'boolean' &&
-		value.notebookUri instanceof URI &&
+	if (
+		(typeof value.defaultKernel === 'object' || value.defaultKernel === undefined) &&
+		(typeof value.connectionProfile === 'object' || value.connectionProfile === undefined) &&
+		value.isDirty() === true &&
```

Comment:
what about opening notebooks that are not dirty?

Reply:
I'm not even sure why you're calling this function here - typically these type guard functions should just be checking that it is the expected interface which shouldn't depend on the actual values of any of the functions on said object.